path/glob.ts

```diff
@@ -1,53 +1,26 @@
 // Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
 // This module is browser compatible.
 
-import { isWindows, type OSType, osType } from "./_os.ts";
-import { SEP, SEP_PATTERN } from "./separator.ts";
-import * as _win32 from "./win32.ts";
-import * as _posix from "./posix.ts";
-
-const path = isWindows ? _win32 : _posix;
-const { join, normalize } = path;
-
-export interface GlobOptions {
-  /** Extended glob syntax.
-   * See https://www.linuxjournal.com/content/bash-extended-globbing.
-   *
-   * @default {true}
-   */
-  extended?: boolean;
-  /** Globstar syntax.
-   * See https://www.linuxjournal.com/content/globstar-new-bash-globbing-option.
-   * If false, `**` is treated like `*`.
-   *
-   * @default {true}
-   */
-  globstar?: boolean;
-  /** Whether globstar should be case-insensitive. */
-  caseInsensitive?: boolean;
-  /** Operating system. Defaults to the native OS. */
+import { isWindows, OSType } from "./_os.ts";
+import type { GlobOptions } from "./_common/glob_to_reg_exp.ts";
+import {
+  globToRegExp as posixGlobToRegExp,
+  joinGlobs as posixJoinGlobs,
+  normalizeGlob as posixNormalizeGlob,
+} from "./posix/glob.ts";
+import {
+  globToRegExp as windowsGlobToRegExp,
+  joinGlobs as windowsJoinGlobs,
+  normalizeGlob as windowsNormalizeGlob,
+} from "./windows/glob.ts";
+
+export { isGlob } from "./_common/is_glob.ts";
+
+export type { GlobOptions };
+
+export type GlobToRegExpOptions = GlobOptions & {
   os?: OSType;
```

Comment:
This was mainly so that the tests still work, personally I think the `os` option should be ditched entirely otherwise it's inconsistent with the other path fns, but the tests will need reworking to account for that.